projects/schematics/src/migrations/5_0/constructor-deprecations/data/cart-lib.migration.ts

```diff
@@ -0,0 +1,2253 @@
+        importPath: '@spartacus/checkout/root',
+      },
+      {
+        className: 'UserIdService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'ConfiguratorUtilsService',
+        importPath: '@spartacus/product-configurator/rulebased',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'ConfiguratorCommonsService',
+    importPath: '@spartacus/product-configurator/rulebased',
+    deprecatedParams: [
+      {
+        className: 'Store',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'CommonConfiguratorUtilsService',
+        importPath: '@spartacus/product-configurator/common',
+      },
+      {
+        className: 'ConfiguratorCartService',
+        importPath: '@spartacus/product-configurator/rulebased',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+      {
+        className: 'ConfiguratorUtilsService',
+        importPath: '@spartacus/product-configurator/rulebased',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'ActiveCartOrderEntriesContext',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActionsSubject',
+        importPath: '@ngrx/store',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ProductImportInfoService',
+        importPath: '@spartacus/cart/base/core',
+      },
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'AddedToCartDialogComponent',
+    importPath: '@spartacus/storefront',
+    deprecatedParams: [
+      {
+        className: 'ModalService',
+        importPath: '@spartacus/storefront',
+      },
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    removeParams: [
+      {
+        className: 'ActiveCartService',
+        importPath: '@spartacus/core',
+      },
+    ],
+    addParams: [
+      {
+        className: 'ActiveCartFacade',
+        importPath: '@spartacus/cart/base/root',
+      },
+    ],
+  },
+  {
+    class: 'AddToCartComponent',
```

Comment:
there are 2 `'AddToCartComponent'`, the next one is also `'AddToCartComponent'`.

Reply:
They are different constructors from the same class.